sdk/core/Azure.Core/src/Internal/ArrayBackedPropertyBag.cs

```diff
@@ -16,31 +16,17 @@ namespace Azure.Core
     /// </summary>
     internal struct ArrayBackedPropertyBag<TKey, TValue> where TKey : struct, IEquatable<TKey>
     {
-        private Kvp _first;
-        private Kvp _second;
-        private Kvp[]? _rest;
+        private (TKey Key, TValue Value) _first;
+        private (TKey Key, TValue Value) _second;
+        private (TKey Key, TValue Value)[]? _rest;
         private int _count;
 #if DEBUG
         private bool _disposed;
 #endif
-        private readonly struct Kvp
-        {
-            public readonly TKey Key;
-            public readonly TValue Value;
-
-            public Kvp(TKey key, TValue value)
-            {
-                Key = key;
-                Value = value;
-            }
-
-            public void Deconstruct(out TKey key, out TValue value)
-            {
-                key = Key;
-                value = Value;
-            }
+        private readonly object _lock = new();
```

Comment:
this lock is not used in release builds. It should be together with the _disposed field (either both should exist in debug builds or both should be absent. And I have a preference for both being there, unless we have evidence that such microptimizations are significant.

Reply:
Will fix

Reply:
The lock _is_ used in release builds so putting back.

Reply:
> And I have a preference for both being there, unless we have evidence that such microptimizations are significant.

Correct me if I'm wrong, but from what I remember, the purpose of this type was to minimize the amount of memory consumed. `_lock` field is always initialized, even when there are no elements in the bag.